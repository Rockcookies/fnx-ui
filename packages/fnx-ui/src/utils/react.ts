import { Children, ReactElement } from 'react';
import { isFragment } from 'react-is';

export const toElementArray = (
	children: React.ReactNode,
	option?: { keepEmpty?: boolean },
): ReactElement[] => {
	const keepEmpty = option && option.keepEmpty;
	let ret: React.ReactElement[] = [];

	Children.forEach(children, (child: any) => {
		if ((child === undefined || child === null) && !keepEmpty) {
			return;
		}

		if (Array.isArray(child)) {
			ret = ret.concat(toElementArray(child));
		} else if (isFragment(child) && child.props) {
			ret = ret.concat(toElementArray(child.props.children, option));
		} else {
			ret.push(child);
		}
	});

	return ret;
};
