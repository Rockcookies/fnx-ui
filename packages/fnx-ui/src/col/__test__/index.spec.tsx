import React from 'react';
import { render } from '@testing-library/react';
import Col from '../index';
import Row from '../../row/index';

describe('<Col>', () => {
	it('should render Col correctly', () => {
		const { container } = render(<Col span={8} />);

		expect(container).toMatchSnapshot();
	});

	it('should render gutter correctly', () => {
		const { container } = render(
			<Row gutter={24}>
				<Col span={24}>24</Col>

				<Col span={12}>12</Col>
				<Col span={12}>12</Col>

				<Col span={8}>8</Col>
				<Col span={8}>8</Col>
				<Col span={8}>8</Col>

				<Col span={6}>6</Col>
				<Col span={6}>6</Col>
				<Col span={6}>6</Col>
				<Col span={6}>6</Col>

				<Col span={7}>7</Col>
				<Col span={6}>6</Col>
				<Col span={5}>5</Col>
				<Col span={4}>4</Col>
				<Col span={3}>3</Col>
				<Col span={2}>2</Col>
			</Row>,
		);

		expect(container).toMatchSnapshot();
	});
});
