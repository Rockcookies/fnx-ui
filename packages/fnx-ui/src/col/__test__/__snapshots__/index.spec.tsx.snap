// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Col> should render Col correctly 1`] = `
<div>
  <div
    class="fnx-col fnx-col--8"
  />
</div>
`;

exports[`<Col> should render gutter correctly 1`] = `
<div>
  <div
    class="fnx-row"
    style="margin-left: -12px; margin-right: -12px;"
  >
    <div
      class="fnx-col fnx-col--24"
      style="padding-left: 12px; padding-right: 12px;"
    >
      24
    </div>
    <div
      class="fnx-col fnx-col--12"
      style="padding-left: 12px; padding-right: 12px;"
    >
      12
    </div>
    <div
      class="fnx-col fnx-col--12"
      style="padding-left: 12px; padding-right: 12px;"
    >
      12
    </div>
    <div
      class="fnx-col fnx-col--8"
      style="padding-left: 12px; padding-right: 12px;"
    >
      8
    </div>
    <div
      class="fnx-col fnx-col--8"
      style="padding-left: 12px; padding-right: 12px;"
    >
      8
    </div>
    <div
      class="fnx-col fnx-col--8"
      style="padding-left: 12px; padding-right: 12px;"
    >
      8
    </div>
    <div
      class="fnx-col fnx-col--6"
      style="padding-left: 12px; padding-right: 12px;"
    >
      6
    </div>
    <div
      class="fnx-col fnx-col--6"
      style="padding-left: 12px; padding-right: 12px;"
    >
      6
    </div>
    <div
      class="fnx-col fnx-col--6"
      style="padding-left: 12px; padding-right: 12px;"
    >
      6
    </div>
    <div
      class="fnx-col fnx-col--6"
      style="padding-left: 12px; padding-right: 12px;"
    >
      6
    </div>
    <div
      class="fnx-col fnx-col--7"
      style="padding-left: 12px; padding-right: 12px;"
    >
      7
    </div>
    <div
      class="fnx-col fnx-col--6"
      style="padding-left: 12px; padding-right: 12px;"
    >
      6
    </div>
    <div
      class="fnx-col fnx-col--5"
      style="padding-left: 12px; padding-right: 12px;"
    >
      5
    </div>
    <div
      class="fnx-col fnx-col--4"
      style="padding-left: 12px; padding-right: 12px;"
    >
      4
    </div>
    <div
      class="fnx-col fnx-col--3"
      style="padding-left: 12px; padding-right: 12px;"
    >
      3
    </div>
    <div
      class="fnx-col fnx-col--2"
      style="padding-left: 12px; padding-right: 12px;"
    >
      2
    </div>
  </div>
</div>
`;
